src/xrc/src/forex.rs

```diff
@@ -938,89 +938,145 @@ impl IsForex for CentralBankOfBosniaHerzegovina {
         )
     }
 
     fn extract_rate(&self, bytes: &[u8], timestamp: u64) -> Result<ForexRateMap, ExtractError> {
         let timestamp = (timestamp / SECONDS_PER_DAY) * SECONDS_PER_DAY;
 
-        let data: XmlBankOfIsraelCurrencies =
-            serde_xml_rs::from_reader(&bytes[3..]).map_err(|_| {
+        let data: XmlBankOfIsraelStructureSpecificData =
+            serde_xml_rs::from_reader(bytes).map_err(|_| {
                 ExtractError::XmlDeserialize(String::from_utf8(bytes.to_vec()).unwrap_or_default())
             })?;
 
-        let values: Vec<&XmlBankOfIsraelCurrency> = data
+        let mut wrong_time = false;
+
+        let data_set_records = data
             .entries
             .iter()
             .filter_map(|entry| match entry {
-                XmlBankOfIsraelCurrenciesOptions::Currency(currency) => Some(currency),
+                XmlBankOfIsraelStructureSpecificDataEnum::DataSet(data_set) => Some(data_set),
                 _ => None,
             })
-            .collect();
+            .collect::<Vec<&XmlBankOfIsraelDataSet>>();
+
+        let data_set = data_set_records.get(0).ok_or_else(|| {
+            ExtractError::XmlDeserialize("Cannot find data set entries".to_string())
+        })?;
 
-        let extracted_timestamp = data
+        let mut values = data_set
             .entries
             .iter()
-            .find(|entry| matches!(entry, XmlBankOfIsraelCurrenciesOptions::LastUpdate(_)))
-            .and_then(|entry| match entry {
-                XmlBankOfIsraelCurrenciesOptions::LastUpdate(s) => Some(
-                    NaiveDateTime::parse_from_str(
-                        &(s.to_string() + " 00:00:00"),
-                        "%Y-%m-%d %H:%M:%S",
-                    )
-                    .unwrap_or_else(|_| NaiveDateTime::from_timestamp(0, 0))
-                    .timestamp() as u64,
-                ),
-                _ => None,
+            .filter_map(|entry| {
+                let quote = &entry.base_currency;
+                let unit = u64::pow(10, entry.unit_mult.parse::<u64>().ok()? as u32);
+                let date = &entry.entries[0].time_period;
+                let value = entry.entries[0].obs_value.parse::<f64>().ok()?;
+
+                let extracted_timestamp = NaiveDateTime::parse_from_str(
+                    &(date.to_string() + " 00:00:00"),
+                    "%Y-%m-%d %H:%M:%S",
+                )
+                .unwrap_or_else(|_| NaiveDateTime::from_timestamp(0, 0))
+                .timestamp() as u64;
+                if extracted_timestamp != timestamp {
+                    wrong_time = true;
+                    return None;
+                }
+                Some((
+                    quote.to_string(),
+                    (value * unit as f64 * RATE_UNIT as f64) as u64,
+                ))
             })
-            .unwrap_or(0);
+            .collect::<ForexRateMap>();
 
-        if extracted_timestamp != timestamp {
+        if wrong_time {
```

Comment:
I thought to make it all-or-nothing for the timestamp, as I do not expect to have values of different days for different currencies (that might mean some concerning inconsistency), but I wanted to be permissive w.r.t parsing of the values of a single currency, so that if for one currency there is an error, we still get the rest. But thinking about it again, any such problem should be as concerning. So I will apply your suggestion.